src/string.js

```diff
@@ -35,3 +36,17 @@ export let autoLabelOption = a => ({
   label: a.label || autoLabel(a.value || a),
 })
 export let autoLabelOptions = _.map(autoLabelOption)
+
+export let toSentenceWith = _.curry((initial, last, join, array) =>
+  _.flow(
+    mapIndexed((x, i) => [
+      x,
+      i === array.length - 1 ? null : i === array.length - 2 ? last : initial,
+    ]),
+    _.flatten,
+    _.compact,
+    join || _.identity
```

Comment:
Join could be `F.arrayToObject`. For React components, join could be a Component, like `children => <div>{children}</div>` :) this way we can wrap each matching element in any style and not have to force ourselves to have a string :) it could also be used for tag components or any kind of component that could resemble a sentence with connectors!!